src/hash_table.c

```diff
@@ -880,11 +886,24 @@ lyht_remove_with_resize_cb(struct hash_table *ht, void *val_p, uint32_t hash, va
             }
 
             /* shrink */
-            ret = lyht_resize(ht, 0);
+            ret = lyht_resize(ht, -1);
 
             if (resize_val_equal) {
                 lyht_set_cb(ht, old_val_equal);
             }
+        } else {
+            if ((ht->size > LYHT_MIN_SIZE) && (ht->size - ht->used - ht->deleted) < (ht->size / 50)) {
```

Comment:
Expression `(ht->size > LYHT_MIN_SIZE)` seems redundant here. Also, refactor the rest of the expression so that the current percentage of `empty` (that is the term used in the code) records is calculated and then compared to a macro that will be set to `2`. Similar to shrinking/enlarging. Finally, this check should be performed every time the hash table is not shrunk (even if `ht->resize` is not `2`). And please commend the code more.